### PR TITLE
Backport nodename changes for OpenStack and GCE to 3.3

### PR DESCRIPTION
**Pull request by detiber**

### CODE REVIEW

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
@sabre1041 completely untested, but this is a starting point.

File: roles/openshift_facts/library/openshift_facts.py [hunk 1]
```diff
@@ -937,6 +941,16 @@ def set_sdn_facts_if_unset(facts, system_facts):
 
     return facts
 
+def set_nodename(facts):
+    if 'node' in facts and 'common' in facts:
+        if 'cloudprovider' in facts and facts['cloudprovider']['kind'] == 'openstack':
+            facts['node']['nodename'] = facts['provider']['metadata']['hostname'].replace('.novalocal', '')
+        elif 'cloudprovider' in facts and facts['cloudprovider']['kind'] == 'gce':
+            facts['node']['nodename'] = '.'.split(facts['provider']['metadata']['hostname'])[0]
```
Comment (sabre1041) [inline roles/openshift_facts/library/openshift_facts.py]: @detiber The location of the hostname value for gce is found in `facts['provider']['metadata']['instance']['hostname']`.

The following line has proven to be successful:
`facts['node']['nodename'] = facts['provider']['metadata']['instance']['hostname'].split('.')[0]`

This issue is also related to #3172

**Comment by mtnbikenc** [comment]
Closing this PR as it is stale and needs to be rebased.   Please reopen the PR if you feel this patch is still applicable.  Thanks!

**Pull request closed without merging**